### DISCUSSION TITLE
Issues report: Gunicorn worker died and webserver grid view bug

### DISCUSSION THREAD

**Discussion by benbenbang** [apache/airflow#24857] [General]
### Apache Airflow version

2.3.2 (latest released)

### What happened

Hello,

1. Issue 1
I am not sure whether the Gunicorn one is a valid question or not. We deploy our Airflow cluster on K8s with
- Webserver * 2
- Scheduler * 3
- Celery worker * 2
- Trigger * 1

Before we upgrade to 2.3, the rolling upgrade can go smoothly without issue. However, after the upgrade, sometimes the webserver's Gunicorn worker can somehow die and **not reboot** by itself. I will need to do some manual trigger or send some signal to kill and reboot it.

2. Issue 2
For the second one, which is about the grid UI,
we've found this is pretty confusing

<img width="608" alt="screenshot_2022-07-05_at_10 31 03" src="https://user-images.githubusercontent.com/6682936/177420473-4a1ad411-4d3a-49ff-b958-3e188a460d08.png">

The max and mean run duration is `hour` in this context, and the min run duration is `minute`
but this looks like the min run duration is longer than max or mean.

Many thanks :pr

### What you think should happen instead

1. The webserver should self-recover and restart by itself
2. The UI should at least with unit or change a way to present the run duration

### Operating System

Debian GNU/Linux 11 (bullseye)

### Versions of Apache Airflow Providers

```
package_name                             | description                                                              | version
=========================================+==========================================================================+========
airflow-provider-great-expectations      | An Apache Airflow provider for Great Expectations                        | 0.1.4
apache-airflow-providers-amazon          | Amazon integration (including Amazon Web Services (AWS)                  | 3.4.0
                                         | https://aws.amazon.com/)                                                 |
apache-airflow-providers-celery          | Celery http://www.celeryproject.org/                                     | 2.1.4
apache-airflow-providers-cncf-kubernetes | Kubernetes https://kubernetes.io/                                        | 4.0.2
apache-airflow-providers-docker          | Docker https://docs.docker.com/install/                                  | 2.7.0
apache-airflow-providers-elasticsearch   | Elasticsearch https://www.elastic.co/elasticsearch                       | 3.0.3
apache-airflow-providers-ftp             | File Transfer Protocol (FTP) https://tools.ietf.org/html/rfc114          | 2.1.2
apache-airflow-providers-google          | Google services including:                                               | 7.0.0
                                         |                                                                          |
                                         |   - Google Ads https://ads.google.com/                                   |
                                         |   - Google Cloud (GCP) https://cloud.google.com/                         |
                                         |   - Google Firebase https://firebase.google.com/                         |
                                         |   - Google LevelDB https://github.com/google/leveldb/                    |
                                         |   - Google Marketing Platform https://marketingplatform.google.com/      |
                                         |   - Google Workspace https://workspace.google.pl/ (formerly Google       |
                                         | Suite)                                                                   |
apache-airflow-providers-hashicorp       | Hashicorp including Hashicorp Vault https://www.vaultproject.io/         | 2.2.0
apache-airflow-providers-http            | Hypertext Transfer Protocol (HTTP) https://www.w3.org/Protocols/         | 2.1.2
apache-airflow-providers-imap            | Internet Message Access Protocol (IMAP)                                  | 2.2.3
                                         | https://tools.ietf.org/html/rfc3501                                      |
apache-airflow-providers-jdbc            | Java Database Connectivity (JDBC)                                        | 2.1.3
                                         | https://docs.oracle.com/javase/8/docs/technotes/guides/jdbc/             |
apache-airflow-providers-mongo           | MongoDB https://www.mongodb.com/what-is-mongodb                          | 2.3.3
apache-airflow-providers-mysql           | MySQL https://www.mysql.com/products/                                    | 2.2.3
apache-airflow-providers-postgres        | PostgreSQL https://www.postgresql.org/                                   | 4.1.0
apache-airflow-providers-redis           | Redis https://redis.io/                                                  | 2.0.4
apache-airflow-providers-salesforce      | Salesforce https://www.salesforce.com/                                   | 3.4.4
apache-airflow-providers-sftp            | SSH File Transfer Protocol (SFTP)                                        | 2.6.0
                                         | https://tools.ietf.org/wg/secsh/draft-ietf-secsh-filexfer/               |
apache-airflow-providers-slack           | Slack https://slack.com/                                                 | 4.2.3
apache-airflow-providers-snowflake       | Snowflake https://www.snowflake.com/                                     | 2.7.0
apache-airflow-providers-sqlite          | SQLite https://www.sqlite.org/                                           | 2.1.3
```

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

- Webserver * 2
- Scheduler * 3
- Celery worker * 2
- Trigger * 1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
First of all please DON't create two issues in one issue. it makes it impossible to resolve and manage. I converted it immediately into discussion because of that. I suggest you start a new issue(s) (if you decide to do it again).

When it comes to 1. you need ot dig deeper and check resource usage and logs of your case when it hangs. It makes impossible to diagnose and help you without more evidences what's happening. Likely you have some resource usage (maybe memory) but that makes it impossible to help you because you can only see it yourself in your deployment and there is nothing we can do to help you and to understand what's going on. Please take a close look at your deployment and look for more evidences. 

for the other issue. Please open a second one exclusively focusing on that one.
